src/app/courses/courses-list/course-item/course-item.component.ts

```diff
@@ -0,0 +1,29 @@
+import { DatePipe } from '@angular/common';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+@Component({
+  selector: 'app-course-item',
+  templateUrl: './course-item.component.html',
+  styleUrls: ['./course-item.component.css']
+})
+export class CourseItemComponent implements OnInit {
+  @Input() title: string = '';
+  @Input() description: string = '';
+  @Input() duration: number | undefined;
+  @Input() creationDate:  Date | undefined;
+  @Input() id: number | undefined;
+  @Output() deletedItemId = new EventEmitter<number>();
+
+  constructor(private datePipe: DatePipe) {}
+
+  formattedDate: string | null = '';
+
+  ngOnInit() {
+    this.formattedDate = this.datePipe.transform(this.creationDate, 'yyyy-MM-dd');
```

Comment:
Very good try and thoughts, but pipes are generally used in html markup - to optimize the build and painting in the browser
You can use it like this in the html  file : duration | date: 'medium'
Please read the link attached 
[DatePipe](https://angular.io/api/common/DatePipe)

Reply:
done, pipes are great things in Angular, very convenient